### PR TITLE
fix: Refactored homepage with new Design System components

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR has updated old design system components with new components. Also, updated colors of gray and texts.

Fixes # (issue)
https://github.com/appsmithorg/appsmith/issues/19386

Media

https://user-images.githubusercontent.com/87797149/231407141-55215a47-24da-49ac-aadb-c40e5b5815ff.mov

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

- Manual
- Jest
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/ce/pages/Applications/index.tsx [hunk 1]
```diff
@@ -802,110 +808,110 @@ export function ApplicationsSection(props: any) {
...
+                                  savingState={
+                                    isSavingWorkspaceInfo
+                                      ? SavingState.STARTED
+                                      : SavingState.NOT_STARTED
+                                  }
+                                  underline
+                                />
+                              </div>
+                              <MenuItem
+                                data-testid="t--workspace-setting"
```
Comment (tanvibhakta) [inline app/client/src/ce/pages/Applications/index.tsx:841]: update relevant cypress tests

File: app/client/src/ce/pages/Home/LeftPaneBottomSection.tsx [hunk 2]
```diff
@@ -106,8 +100,19 @@ function LeftPaneBottomSection() {
         }}
         text={createMessage(WELCOME_TOUR)}
       />
-
-      <ProductUpdatesModal />
+      <MenuItem
```
Comment (tanvibhakta) [inline app/client/src/ce/pages/Home/LeftPaneBottomSection.tsx:103]: Why are we using old menu here?

File: app/client/src/pages/Applications/ApplicationCard.tsx [hunk 3]
```diff
@@ -256,13 +231,13 @@ const Control = styled.div<{ fixed?: boolean }>`
...
 `;
 
-const MoreOptionsContainer = styled.div`
-  width: 22px;
-  height: 22px;
-  display: flex;
-  align-items: center;
-  justify-content: center;
-`;
+// const MoreOptionsContainer = styled.div`
```
Comment (tanvibhakta) [inline app/client/src/pages/Applications/ApplicationCard.tsx:234]: remove if unused?

File: app/client/yarn.lock [hunk 4]
```diff
@@ -8373,7 +8373,7 @@ babel-plugin-react-docgen@^4.1.0, babel-plugin-react-docgen@^4.2.1:
     lodash "^4.17.15"
     react-docgen "^5.0.0"
 
-"babel-plugin-styled-components@>= 1.12.0":
+"babel-plugin-styled-components@>= 1.12.0", babel-plugin-styled-components@^2.0.7:
```
Comment (tanvibhakta) [inline app/client/yarn.lock:8376]: remove and regenerate

**Review by tanvibhakta — CHANGES_REQUESTED** [review]

File: app/client/src/ce/pages/Applications/index.tsx [hunk 5]
```diff
@@ -376,7 +382,7 @@ export function WorkspaceMenuItem({
   }, [selected]);
 
   return (
-    <MenuItem
+    <ListItem
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/src/ce/pages/Applications/index.tsx:385]: There is no list item component in new ADS... this is not a menu.

File: app/client/src/ce/pages/Applications/index.tsx [hunk 6]
```diff
@@ -777,21 +791,13 @@ export function ApplicationsSection(props: any) {
...
-                          setWarnDeleteWorkspace(false);
-                        }}
-                        position={Position.BOTTOM_RIGHT}
-                        target={
+                        data-testid="t--workspace-name"
+                        // disabled={isFetchingApplications}
+                        onOpenChange={handleWorkspaceMenuClose}
+                        open={workspace.id === workspaceToOpenMenu}
+                      >
+                        <MenuTrigger>
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/src/ce/pages/Applications/index.tsx:800]: Good catch

File: app/client/src/ce/pages/Applications/index.tsx [hunk 7]
```diff
@@ -777,21 +791,13 @@ export function ApplicationsSection(props: any) {
...
-                        onClose={() => {
-                          setWorkspaceToOpenMenu(null);
-                        }}
-                        onClosing={() => {
-                          setWarnLeavingWorkspace(false);
-                          setWarnDeleteWorkspace(false);
-                        }}
-                        position={Position.BOTTOM_RIGHT}
-                        target={
+                        data-testid="t--workspace-name"
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/src/ce/pages/Applications/index.tsx:795]: That is in a separate PR

File: app/client/src/pages/Applications/ApplicationCard.tsx (hunk 3 above)
Comment (albinAppsmith) [reply app/client/src/pages/Applications/ApplicationCard.tsx:234]: Done

File: app/client/craco.dev.config.js [hunk 8]
```diff
@@ -3,6 +3,9 @@ const { merge } = require("webpack-merge");
 const common = require("./craco.common.config.js");
 
 module.exports = merge(common, {
+  babel: {
+    plugins: ["babel-plugin-styled-components"],
+  },
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/craco.dev.config.js:8]: Done

File: app/client/package.json [hunk 9]
```diff
@@ -255,6 +255,7 @@
     "@typescript-eslint/eslint-plugin": "^5.25.0",
     "@typescript-eslint/parser": "^5.25.0",
     "babel-plugin-module-resolver": "^4.1.0",
+    "babel-plugin-styled-components": "^2.0.7",
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/package.json:261]: Done

File: app/client/src/ce/pages/Applications/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (tanvibhakta) [reply app/client/src/ce/pages/Applications/index.tsx:385]: Ah my bad

**Review by tanvibhakta — APPROVED** [review]

**Comment by vasanth-appsmith** [comment]
https://user-images.githubusercontent.com/92305123/231458625-c8df0989-d612-4235-b1d3-1beb869ad4db.mov

1. Menu opens and closed with a 2-3 sec delay.

**Pull request merged**